File: loja/clientes/src/main/java/br/com/bota/service/impl/ClienteServiceImpl.java
package br.com.bota.service.impl;

import br.com.bota.entity.Cliente;
import br.com.bota.repository.ClienteRepository;
import br.com.bota.service.ClienteService;
import br.com.bota.service.rule.CustomerNotFoundException;

import javax.inject.Inject;
import javax.inject.Singleton;
import javax.transaction.Transactional;
import javax.validation.Valid;
import java.util.List;
import java.util.UUID;

@Singleton
public class ClienteServiceImpl implements ClienteService {
    @Inject
    private ClienteRepository repository;


    @Transactional
    @Override
    public Cliente save(
            @Valid Cliente customer) {
        if (customer.getId() == null) {
            return repository.save(customer);
        } else {
            var existent = repository.findById(customer.getId());
            existent.ifPresent(c -> c.setNome(customer.getNome()));
            return customer;
        }
    }

    @Override
    public Cliente findById(UUID id) {
        return repository.findById(id).orElseThrow(() -> new CustomerNotFoundException());
    }

    @Override
    public List<Cliente> findAll() {
        return repository.findAll();
    }
}

File: loja-kotlin/kclientes/src/main/kotlin/br/com/bota/service/impl/ClienteServiceImpl.kt
package br.com.bota.service.impl

import br.com.bota.entity.Cliente
import br.com.bota.exceptions.rule.CustomerNotFoundException
import br.com.bota.repository.ClienteRepository
import br.com.bota.service.ClienteService
import java.util.*
import jakarta.inject.Inject
import jakarta.inject.Singleton
import javax.transaction.Transactional
import javax.validation.Valid

@Singleton
open class ClienteServiceImpl(
    @Inject
    private val repository: ClienteRepository
) : ClienteService {

    override fun findAll(): List<Cliente> {
        return repository.findAll()
    }

    @Transactional
    override fun save(@Valid cliente: Cliente): Cliente {
        return if (cliente.id === null) {
            repository.save(cliente)
        } else {
            val existent = findById(cliente.id!!)
            existent.nome = cliente.nome
            existent
        }
    }

    override fun findById(id: UUID): Cliente {
        return repository.findById(id).orElseThrow { CustomerNotFoundException() }
    }
}